src/utils/callDockerode.ts

```diff
@@ -9,12 +9,15 @@ import { IActionContext, parseError, UserCancelledError } from 'vscode-azureexte
 import { ext } from '../extensionVariables';
 import { localize } from '../localize';
 import { dockerContextManager } from './dockerContextManager';
-import { CancellationPromiseSource } from './promiseUtils';
+import { CancellationPromiseSource, TimeoutPromiseSource } from './promiseUtils';
 import { refreshDockerode } from './refreshDockerode';
 
+// 20 s timeout for all calls (enough time for a possible Dockerode refresh + the call, but short enough to be UX-reasonable)
+const dockerodeCallTimeout = 20 * 1000;
+
 let cps: CancellationPromiseSource | undefined;
 
-export async function callDockerode<T>(dockerodeCallback: () => T): Promise<T> {
+export async function callDockerode<T>(dockerodeCallback: () => T, context?: IActionContext): Promise<T> {
```

Comment:
The caller should have an ability to pass the timeout. Not all calls require 20 seconds timeout

Reply:
All calls can result in a Dockerode refresh, so it makes sense to use the same timeout.

Reply:
I don't understand your comment. What is the relation between any dockerode call to dockerode refresh? Say if I want to issue an inspect command with 5 seconds time out, how do I do that?

Reply:
The point of wrapping everything in `callDockerode` is so that we can detect context changes and re-initialize the Dockerode object if a change is detected. Ideally, Dockerode would do this internally, but that isn't the case.

Any Dockerode call (including inspect) _could_ be the first one to detect a context switch, in which case we cancel all outstanding Dockerode calls, refresh Dockerode, then return. Detecting requires about 25 ms (hash check of the config file), then reinitializing Dockerode takes ~1-6 seconds depending on how long `docker context inspect` feels like taking.

Reply:
Yes the context refresh happens when the context is changed. My suggestion is to take timeout as optional parameters and use it when provided.

Reply:
That's reasonable, but there isn't anywhere I'd actually use it right now. Can we just defer adding that option until we actually find that we need it?

Reply:
Since now this supports timeout, it is reasonable to expose the timeout to the caller. I don't think it is much work to add an optional parameter. In any case, it is your call.

Reply:
I agree that it would be very easy to add an optional parameter to override the default. However, since there isn't currently a use case, I'm OK adding that parameter if/when we need it.